tests/PolyType.Tests/AssociatedTypesTests.cs

```diff
@@ -49,17 +55,25 @@ public void CanConstructAssociatedType_ByExtension()
     {
         ITypeShape? typeShape = providerUnderTest.Provider.GetShape(typeof(GenericDataType<int, string>));
         Assert.NotNull(typeShape);
-        Func<object>? factory = typeShape.GetAssociatedTypeFactory(typeof(GenericDataTypeVerifier<,>));
+        IObjectTypeShape? associatedShape = (IObjectTypeShape?)typeShape.GetAssociatedTypeShape(typeof(GenericDataTypeVerifier<,>));
+        Assert.NotNull(associatedShape);
+        Func<object>? factory = associatedShape.GetDefaultConstructor();
         Assert.NotNull(factory);
         Assert.IsType<GenericDataTypeVerifier<int, string>>(factory.Invoke());
+
+        if (partialShapesSupported)
+        {
+            // Verify the associated type's shape is only partially available.
+            Assert.Empty(associatedShape.Properties);
```

Comment:
Should querying for properties on a partial shape throw or merely return an empty collection? I'm thinking throwing may be more appropriate to avoid confusion when collections are empty when they otherwise wouldn't be. But if we throw, we should probably add a depth property to the shape so that a program that cares may avoid an exception while exploring arbitrary shapes.

Reply:
Good question. I think throwing is appropriate here (particularly if the reflection-based typeshape of the same type enumerates the properties correctly).